### PR TITLE
Tasks

### PR DESCRIPTION
**Pull request by chenyuqi990215**
1. add classification and regression tasks.
2. add inverse mapping in base model.
3. fix typo in base dataset.

### CODE REVIEW

File: physiopro/dataset/detection.py [hunk 1]
```diff
@@ -1,87 +1,35 @@
...
-import numpy as np
 from .preprocessor import Preprocessor
-from .base import DATASETS, BaseDataset
+from .base import DATASETS
+from .classification import ClassificationDataset
 
 
 @DATASETS.register_module("detection")
-class DetectionDataset(BaseDataset):
+class DetectionDataset(ClassificationDataset):
```
Comment (suiguoxin) [inline physiopro/dataset/detection.py:12]: Can we remove this class?

File: physiopro/dataset/regression.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+from utilsd.config import ClassConfig
+import pandas as pd
+import numpy as np
+from .preprocessor import Preprocessor
+from .base import DATASETS, BaseDataset
+
+
+@DATASETS.register_module("regression")
+class RegressionDataset(BaseDataset):
+    def __init__(
```
Comment (suiguoxin) [inline physiopro/dataset/regression.py:14]: Please add a document, just list all the dataset types available and describe briefly what they are designed for.

File: physiopro/dataset/detection.py (hunk 1 above)
Comment (chenyuqi990215) [reply physiopro/dataset/detection.py:12]: if remove, we probably need to add a parameter in classification dataset to identify if the task is binary detection task.

File: physiopro/dataset/regression.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        self,
+        data_folder: Path,
+        meta_path: Path,
+        num_variables: int = 18,
+        freq: int = 256,
+        max_seq_len: int = 30,
+        num_classes: int = 2,
```
Comment (suiguoxin) [inline physiopro/dataset/regression.py:21]: `num_classes` is redundant for regression

File: physiopro/dataset/detection.py (hunk 1 above)
Comment (suiguoxin) [reply physiopro/dataset/detection.py:12]: > if remove, we probably need to add a parameter in classification dataset to identify if the task is binary detection task.

`num_classes` is by default 2, isn't that enough?

File: physiopro/dataset/classification.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    def get_index(self):
+        assert self.label is not None and self.feature is not None
+        return self.label_index
+
+    def get_label(self, pred):
+        if self.inv_label_mapping is not None:
+            indices = pred.argmax(axis=-1)
+            return np.array([self.inv_label_mapping[idx] for idx in indices])
+        else:
```
Comment (suiguoxin) [inline physiopro/dataset/classification.py:104]: Unnecessary `else` with return

File: physiopro/dataset/detection.py (hunk 1 above)
Comment (chenyuqi990215) [reply physiopro/dataset/detection.py:12]: fixed.

File: physiopro/dataset/regression.py (hunk 3 above)
Comment (chenyuqi990215) [reply physiopro/dataset/regression.py]: fixed.

File: physiopro/dataset/classification.py (hunk 4 above)
Comment (chenyuqi990215) [reply physiopro/dataset/classification.py]: fixed.

File: physiopro/dataset/regression.py (hunk 2 above)
Comment (chenyuqi990215) [reply physiopro/dataset/regression.py:14]: Currently, I'm not sure what is regression designed for.

File: docs/configs/transformer_tusz_v2.0.0.yml [hunk 5]
```diff
@@ -1,10 +1,11 @@
 data:
-  type: detection
+  type: classification
```
Comment (suiguoxin) [inline docs/configs/transformer_tusz_v2.0.0.yml:2]: Please also check the data type of other config files in this folder

File: physiopro/dataset/detection.py (hunk 1 above)
Comment (suiguoxin) [reply physiopro/dataset/detection.py:12]: I mean can we remove the class `DetectionDataset`? It's almost same as `ClassificationDataset` and I feel the concept of `detection` confusing.

**Review by suiguoxin — APPROVED** [review]

**Pull request merged**